core.js

```diff
@@ -105,69 +68,90 @@ function applySubraceBonuses(char){
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 140 + " lbs";
-            break;
-        case race === "halfling":
-            additionalHeightAndWeight = returnHeightAndWeight("2d4", "1d1");
-            heightAndWeight[0] = additionalHeightAndWeight[0] + 2.7 + " ft";
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 35 + " lbs";
-            break;
-        case race === "human":
-            additionalHeightAndWeight = returnHeightAndWeight("2d10", "2d4");
-            heightAndWeight[0] = additionalHeightAndWeight[0] + 4.8 + " ft";
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 110 + " lbs";
-            break;
-        case race === "tiefling":
-            additionalHeightAndWeight = returnHeightAndWeight("2d8", "2d4");
-            heightAndWeight[0] = additionalHeightAndWeight[0] + 4.9 + " ft";
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 110 + " lbs";
-            break;
-
-        default: return "invalid race provided as function arguement [returnCharacterHeight()]";
-
-
-    }
-
-    return heightAndWeight;
-
+/**-------------------------------------------------------------------------------
+ * @description Calculates character `height`. The calculation adds a 'base height' 
+ * with a dice throw specific to each race. The base values are retrieve from:
+ *  https://www.reddit.com/r/DnD/comments/abebq0/chart_with_heightweight_ranges_and_averages_for/ 
+ * @param {String} race - the character race.
+ * @returns {Number} the calculated height in ft.  
+*///------------------------------------------------------------------------------
+function calculateCharacterHeight(race) {
+  let finalHeight = 0.0;
+  switch (race) {
+      case "dragonborn":
+          finalHeight = 5.6 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "dwarf":
+          finalHeight = 3.8 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "elf":
+          finalHeight = 4.6 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "gnome":
+          finalHeight = 2.11 + parseFloat((utils.diceThrow(2,4)/ 12).toFixed(1));
+      break;
+      case "half-elf":
+          finalHeight = 4.9 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "half-orc":
+          finalHeight = 4.10 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "halfling":
+          finalHeight = 2.7 + parseFloat((utils.diceThrow(2,4)/ 12).toFixed(1));
+      break;
+      case "human":
+          finalHeight = 4.8 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "tiefling":
+          finalHeight = 4.9 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      default: return "invalid race provided as function arguement [returnCharacterHeight()]";
+  }
+  return finalHeight;
```

Comment:
I have just tested this and in some instances the finalHeight isn't fixed to 1 decimal place

![image](https://user-images.githubusercontent.com/36999273/69576703-c4eb8f00-0fc4-11ea-9dac-dba476279a9d.png)

I assume the total result of the addition of the base height and the randomized amount needs to be  set to .toFixed(1)

Reply:
Yes, this is correct, I will also add this change to the next PR